code/modules/hydroponics/hydroponics.dm

```diff
@@ -521,6 +521,12 @@
 		mutmod = 0
 		adjustNutri(round(S.get_reagent_amount("robustharvestnutriment") *1 ))
 
+
+	//Fish Water is both an excellent fertilizer and waters
+	if(S.has_reagent("fishwater", 1))
```

Comment:
Wasn't the table approach old botany used a pretty good approach for handling many generic chems easily, without a big old nasty if chain